### PR TITLE
Use filter UINT where only positive integers are allowed

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #40116 .

### Summary of Changes
We have a filter="integer" which allows entering negative values into a number field. In featured blog layout this causes an error. 
Suggestion: 
Use filter="UINT" where only positive integer values are allowed.

This would make sense everywhere where only positive values are allowed and a user enters negative numbers only accidentally. 

### Testing Instructions
see #40116. 

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by joomdonation** [comment]
Use `UINT` is also OK. But please also add **min=0** to prevent users enter a negative value into the field and has it magically changed to a positive value.

Also, please make similar changes to category view and also in component options, too.

**Comment by brianteeman** [comment]
> Use UINT is also OK. But please also add min=0 to prevent users enter a negative value into the field and has it magically changed to a positive value.

Just tried that and it still changes -2 to 2 on save

**Comment by joomdonation** [comment]
> Just tried that and it still changes -2 to 2 on save

Ah, Yes. I tried to use the up/down arrow to change the value and it does not allow value smaller than 0. But it does not prevent entering a negative value directly on the field. So in this case, I think we should use number validation as I suggested in the issue report. In case users entered a wrong value for some reasons, they know that it is not allowed

```xml
validate="number"
min="0"
```

**Comment by brianteeman** [comment]
>  In case users entered a wrong value for some reasons, they know that it is not allowed

Definitely. It's never a good idea to change entered data without notification.

**Comment by joomdonation** [comment]
@chmst Please don't remove `filter="integer"`, just leave the filter there as how it is.

File: components/com_content/tmpl/featured/default.xml [hunk 1]
```diff
@@ -35,6 +35,8 @@
 				type="number"
 				label="JGLOBAL_NUM_LEADING_ARTICLES_LABEL"
 				filter="integer"
+				validate="number"
+                min="0"
```
Comment (joomdonation) [inline components/com_content/tmpl/featured/default.xml]: @chmst Could you please check this and other similar lines below ? Need more indentation here.

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 414c74c0f2c72db196c203756686a610a65806bb<br /><br />Tested successfully in Joomla 4.3.0-beta5-dev of 16 March using PHP 8.1.10<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40120">issues.joomla.org/tracker/joomla-cms/40120</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 414c74c0f2c72db196c203756686a610a65806bb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40120">issues.joomla.org/tracker/joomla-cms/40120</a>.</sub>

File: components/com_content/tmpl/featured/default.xml (hunk 1 above)
Comment (joomdonation) [reply components/com_content/tmpl/featured/default.xml]: @chmst Could you please check my last comment? Just minor changes and this PR will be ready to merge.

Comment (chmst) [reply components/com_content/tmpl/featured/default.xml]: done, thx

**Comment by joomdonation** [comment]
RTC. There were two success tests before and the two last commits were just some code style change.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40120">issues.joomla.org/tracker/joomla-cms/40120</a>.</sub>

File: components/com_content/tmpl/featured/default.xml [hunk 2]
```diff
@@ -53,6 +55,8 @@
 				type="number"
 				label="JGLOBAL_NUM_INTRO_ARTICLES_LABEL"
 				filter="integer"
+				validate="number"
+				 min="0"
```
Comment (Quy) [inline components/com_content/tmpl/featured/default.xml]:
```suggestion
				min="0"
```

**Comment by obuisard** [comment]
Thank you Christiane @chmst for the PR!

**Pull request merged**